src/runtime_src/core/include/experimental/xrt_elf.h

```diff
@@ -32,7 +33,17 @@ class elf : public detail::pimpl<elf_impl>
 public:
   XRT_API_EXPORT
   elf(const std::string& fnm);
-  
+
+  /**
+   * elf() - Constructor from raw ELF data
+   *
+   * @param data
+   *  Raw data of elf
+   *
+   */
+  XRT_API_EXPORT
+  elf(const std::vector<char>& data);
```

Comment:
A constructor from std::vector isn't great IMO.   What if data is not held in a vector?   A std::span would be better, but that's c++20.  I would opt for `const char*, size_t` but you are still faced with the problem of how to create the istream without copying the in-memory buffer.

Reply:
@stsoe 
Updated that PR with changes discussed offline.